# GOOGLE DATA ANALYTICS: CAPSTONE
## Case Study 2: Bellabeat, How Can a Wellness Technology Company Play It Smart?
Autor: Omar Ventocilla Lázaro<br>Date: 9/9/2021
    
---
### ASK
Bellabeat, a high-tech company that manufactures health-focused smart products, wants to unlock new growth opportunities for the company. So, I will provide high-level recommendations for the marketing strategy. In this case study, as a junior data analyst, I will analyze data of the wellness watch using a Kaggle dataset for FitBit Fitness Tracker (CC0: Public Domain license). This dataset has 18 csv files, it was generated by respondents to a distributed survey on 2016 (physical activity, heart rate, sleep monitoring, etc).

It is necessary to analyze the data collected from the customers and see how they use smart devices in order to extract information about trends and patterns that exist within the data, and learn how those trends can help Bellabeat's marketing strategy. The discoveries, trends and patterns that are found can be used to learn how most customers might use Bellabeat products. With this knowledge, it will be possible to decide the necessary strategic actions to be taken in order to reach more people.

**GOALS**:
* Reveal more opportunities for growth
* Gain insight into how people are already using smart devices
* Find trends and relationships
* Give high-level recommendations


---
### PREPARE
I will work with some of the 18 files, so first I need to make sure that the data does not have any problem (bias, duplicates, coherence and consistency). 


In [1]:
import pandas as pd
df_activity = pd.read_csv('Case2/dailyActivity_merged.csv')
df_activity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

This file show us the daily user activity (total steps, total distance, sedentary minutes, calories).

In [2]:
df_activity.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  
count   940.000000  
mean   2303.609574  
std     718.166862  
min       0.000000  
25%    1828.500000  
50%    2134.000000  
75%    2793.250000  
max    4900.000000

In [3]:
df_activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

In [4]:
print(df_activity[df_activity.duplicated()]) #Empty

Empty DataFrame
Columns: [Id, ActivityDate, TotalSteps, TotalDistance, TrackerDistance, LoggedActivitiesDistance, VeryActiveDistance, ModeratelyActiveDistance, LightActiveDistance, SedentaryActiveDistance, VeryActiveMinutes, FairlyActiveMinutes, LightlyActiveMinutes, SedentaryMinutes, Calories]
Index: []


The values of these data are consistency, they are greater than cero, there isn't any null value and any duplicated row, all the rows have the correct data type.

There might be a problem with this file because when some user was not active, the 'SedentaryMinutes' should be 1440 minutes (24 hours), but 5 rows have other values. It seems like the smart watch registered only a few hours.

In [5]:
df_activity[df_activity['TotalSteps'] == 0].iloc[:,-2].value_counts().to_frame()

SedentaryMinutes
1440                72
48                   1
966                  1
711                  1
111                  1
1407                 1

---

In [6]:
df_hourlysteps = pd.read_csv('Case2/hourlySteps_merged.csv')
df_hourlysteps.head()

Id           ActivityHour  StepTotal
0  1503960366  4/12/2016 12:00:00 AM        373
1  1503960366   4/12/2016 1:00:00 AM        160
2  1503960366   4/12/2016 2:00:00 AM        151
3  1503960366   4/12/2016 3:00:00 AM          0
4  1503960366   4/12/2016 4:00:00 AM          0

In [7]:
df_hourlysteps[df_hourlysteps.duplicated()] #Empty

Empty DataFrame
Columns: [Id, ActivityHour, StepTotal]
Index: []

This file show us the total steps per hour.

In [8]:
df_hourlysteps.describe() #Greater than cero

Id     StepTotal
count  2.209900e+04  22099.000000
mean   4.848235e+09    320.166342
std    2.422500e+09    690.384228
min    1.503960e+09      0.000000
25%    2.320127e+09      0.000000
50%    4.445115e+09     40.000000
75%    6.962181e+09    357.000000
max    8.877689e+09  10554.000000

In [9]:
df_hourlysteps.info() #Correct data type, any null value

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22099 entries, 0 to 22098
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            22099 non-null  int64 
 1   ActivityHour  22099 non-null  object
 2   StepTotal     22099 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 518.1+ KB


Everything seems ok

In [10]:
df_hourlysteps.Id.value_counts()

1927972279    736
1624580081    736
2022484408    736
2873212765    736
2026352035    736
4558609924    736
2320127002    735
8378563200    735
4388161847    735
8877689391    735
8053475328    735
4445114986    735
7086361926    733
6962181067    732
4020332650    732
4702921684    731
1844505072    731
5553957443    730
4319703577    724
8583815059    718
1503960366    717
5577150313    708
1644430081    708
3977333714    696
8792009665    672
6290855005    665
6117666160    660
6775888955    610
7007744171    601
3372868164    472
8253242879    431
2347167796    414
4057192912     88
Name: Id, dtype: int64

Most files do not have the same number of rows per user.

---
### PROCESS

In [11]:
df_sleepday = pd.read_csv('Case2/sleepDay_merged.csv')
df_sleepday.head()

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [12]:
df_sleepday[df_sleepday.duplicated()] #Not Empty

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
161  4388161847   5/5/2016 12:00:00 AM                  1                 471   
223  4702921684   5/7/2016 12:00:00 AM                  1                 520   
380  8378563200  4/25/2016 12:00:00 AM                  1                 388   

     TotalTimeInBed  
161             495  
223             543  
380             402

In [13]:
df_sleepday.describe() #Greater than cero, 410 rows for each column

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed
count  4.130000e+02         413.000000          413.000000      413.000000
mean   5.000979e+09           1.118644          419.467312      458.639225
std    2.060360e+09           0.345521          118.344679      127.101607
min    1.503960e+09           1.000000           58.000000       61.000000
25%    3.977334e+09           1.000000          361.000000      403.000000
50%    4.702922e+09           1.000000          433.000000      463.000000
75%    6.962181e+09           1.000000          490.000000      526.000000
max    8.792010e+09           3.000000          796.000000      961.000000

In [14]:
df_sleepday.info() #Correct data type, any null value

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB


---
### ANALYZE

In [15]:
#Removing duplicates
df_sleepday = df_sleepday.drop_duplicates()

In [16]:
df_hourlysteps.head()

Id           ActivityHour  StepTotal
0  1503960366  4/12/2016 12:00:00 AM        373
1  1503960366   4/12/2016 1:00:00 AM        160
2  1503960366   4/12/2016 2:00:00 AM        151
3  1503960366   4/12/2016 3:00:00 AM          0
4  1503960366   4/12/2016 4:00:00 AM          0

We need to transform the column 'ActivityHour' to see the average of this value for each hour.

In [17]:
for i in range(df_hourlysteps.shape[0]):
    index = df_hourlysteps.iloc[i,1][:-3].rfind(' ')
    df_hourlysteps.iloc[i,1] = df_hourlysteps.iloc[i,1][index+1:]
steps = df_hourlysteps.groupby(['ActivityHour'], as_index=False).mean()

In [18]:
steps.head()

ActivityHour            Id   StepTotal
0  10:00:00 AM  4.844647e+09  481.665231
1  10:00:00 PM  4.853337e+09  237.987832
2  11:00:00 AM  4.848036e+09  456.886731
3  11:00:00 PM  4.849572e+09  122.132890
4  12:00:00 AM  4.846554e+09   42.188437

We need to transform the column 'ActivityHour' to see the average of this value for each hour.

In [21]:
df_hourlycalories = pd.read_csv('Case2/hourlyCalories_merged.csv')

for i in range(df_hourlycalories.shape[0]):
    index = df_hourlycalories.iloc[i,1][:-3].rfind(' ')
    df_hourlycalories.iloc[i,1] = df_hourlycalories.iloc[i,1][index+1:]
calories = df_hourlycalories.groupby(['ActivityHour'], as_index=False).mean()

In [23]:
df_sleepday.describe().iloc[:2,1:] #Average time asleep

TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed
count         410.000000          410.000000      410.000000
mean            1.119512          419.173171      458.482927

In [24]:
df_activity['TotalSteps'].mean() #Average daily steps 

7637.9106382978725

---
### SHARE

In [ ]:
#Data visualization
import matplotlib.pyplot as plt
import numpy as np

df_mean_day = df_activity.groupby(['ActivityDate'], as_index=False).mean()
df_mean_day = pd.concat([df_mean_day.iloc[:20,:], df_mean_day.iloc[23:,:], df_mean_day.iloc[20:23,:]])
plt.figure(figsize=(40,10))
plt.plot(df_mean_day.iloc[:,0], df_mean_day.iloc[:,2], c='red', label='Average Steps for each day')
plt.plot(df_mean_day.iloc[:,0], np.full((31,1),df_activity['TotalSteps'].mean()), label='Total Average') #average
plt.legend(fontsize=20)
plt.xlabel('Days April-May 2016', fontsize=25)
plt.ylabel('Average steps', fontsize=25)
plt.show()

In [ ]:
index = np.array([4,6,8,10,12,14,16,18,20,22,0,2,5,7,9,11,13,15,17,19,21,23,1,3])

plt.figure(figsize=(40,10))
plt.plot(steps.iloc[index,].iloc[:,0], steps.iloc[index,].iloc[:,2], c='lime', label='Average Steps for each hour')
plt.plot(steps.iloc[index,].iloc[:,0], np.full((24,1), steps['StepTotal'].mean()), c='blue', label='Total Average') #average
plt.legend(fontsize=20)
plt.xlabel('Day hour', fontsize=25)
plt.ylabel('Average Steps', fontsize=25)
plt.show()

In [ ]:
plt.figure(figsize=(40,10))
plt.plot(calories.iloc[index,].iloc[:,0], calories.iloc[index,].iloc[:,2], c='red', label='Calories burned for each hour')
plt.legend(fontsize=20)
plt.xlabel('Day hour', fontsize=25)
plt.ylabel('Average Calories burned', fontsize=25)
plt.show()

In [ ]:
import seaborn as sns

df_var = pd.concat([calories.iloc[:,0:3:2], steps.iloc[:,0:3:2]], axis=1)
plt.figure(figsize=(7,7))
plt.title('Correlation between steps and calories', fontsize=15)
sns.heatmap(df_var.iloc[:,1::2].corr(),vmin=0.9,vmax=1, square= True, annot=True)

**CONCLUSIONS**

* The data was logged for 31 days.
* Average time asleep is 419.1 minutes ≃ 7 hours.
* Users do not sleep enough time, they should sleep almost 8 hours in order to have a healthy life.
* The average steps per day is 7637 (approximately 19,028 ft / 5.8 kilometers).
* 6:00 - 7:00 pm is the period where users are more active.
* The more steps you take, the more calories you burn.

Bellabeat app should recommend to the user to sleep at least 8 hours, it needs to gather more data about sleeping activity and calories consumption. Bellabeat should also recommend to the user to drink some water when they have taken approximately 1000 steps. The app should notify the users when they reach a certain value of calories burned. Besides, the company should gather more further data through more diverse method in order to explore different topics, taking into consideration the day and the age/gender.